File: two-sum/two_sum_test.py
import unittest

from two_sum import twoSum

class TestTwoSum(unittest.TestCase):
  def setUp(self):
    self.testCases = [
      {
        'nums': [2, 7, 11, 15],
        'target': 9,
        'want': [0, 1],
      },
      {
        'nums': [3, 2, 4],
        'target': 6,
        'want': [1, 2],
      },
      {
        'nums': [3, 3],
        'target': 6,
        'want': [0, 1],
      },
      {
        'nums': [-1, 10, 9, 7],
        'target': 6,
        'want': [0, 3],
      },
      {
        'nums': [0, 0, 10, 12],
        'target': 0,
        'want': [0, 1],
      },
      {
        'nums': [2, 5, 5, 11],
        'target': 10,
        'want': [1, 2],
      },
    ]

  def test_want(self):
    for testCase in self.testCases:
      with self.subTest(testCase=testCase):
        self.assertEqual(twoSum(testCase['nums'], testCase['target']), testCase['want'])

if __name__ == '__main__':
  unittest.main()

File: two-sum/two_sum_test.go
package twoSum

import "testing"

func TestTwosum(t *testing.T) {
	testCases := []struct {
		nums   []int
		target int
		want   []int
	}{
		{[]int{2, 7, 11, 15}, 9, []int{0, 1}},
		{[]int{3, 2, 4}, 6, []int{1, 2}},
		{[]int{3, 3}, 6, []int{0, 1}},
		{[]int{-1, 10, 9, 7}, 6, []int{0, 3}},
		{[]int{0, 0, 10, 12}, 0, []int{0, 1}},
	}

	for idx, testCase := range testCases {
		got := twoSum(testCase.nums, testCase.target)
		if !(compareArrays(got, testCase.want)) {
			t.Errorf("%d fali: expected is {%d, %d}, but actually {%d, %d}", idx, testCase.want[0], testCase.want[1], got[0], got[1])
		}
	}
}

func compareArrays(a []int, b []int) bool {
	if (a[0] == b[0] && a[1] == b[1]) || (a[1] == b[0] && a[0] == b[1]) {
		return true
	}
	return false
}
